docs/devguide.md

```diff
@@ -182,6 +182,25 @@ _staging_ ends.
 
 All `Conditions` methods are idempotent and most support _varargs_.
 
-# Plan Controller
+## User Experience
+
+The impact of new changes in MTC on the overall user experience of migrations _must_ be taken into account. 
+
+### Progress Reporting
+
+Migrations in production environments can take significantly long amount of time due to the huge scale of deployed resources. The easiest way to provide better user experience in such cases is by making the migration process transparent to the end user. Migration controller provides a way to report information about ongoing work back to the user in the status field of _MigMigration_ CR in the form of progress messages. Consider leveraging this existing progress reporting mechanism to improve visibility into the migration process.
```

Comment:
Suggest 
`Migrations in production environments can take significantly long amount of time` 
-> 
`Migrations in production environments can a significant amount of time`

Reply:
It might be worth mentioning here that mig-ui is configured to read pipeline progress from `t.SetProgress` calls, one less UI integration piece to worry about.